### ISSUE TITLE
[15.3.x][next-app] Unable to import React/Workspace library

### ISSUE THREAD

**Issue by atanaskanchev** [nrwl/nx#13790]
### Current Behavior

Unable to import generated components into the app `Cannot find module '@my-app/test-lib' or its corresponding type declarations.`

### Expected Behavior

The generated components should be importable by their paths mapping

`tsconfig.base.json`
```
"paths": {
      "@my-app/test-lib": ["libs/test-lib/src/index.ts"],
      "@my-app//ui/web-app/component/auth": [
        "libs/ui/web-app/component/auth/src/index.ts"
      ]
    }`

### Github Repo

_No response_

### Steps to Reproduce

1. Create a new workspace `npx create-nx-workspace@latest --allPrompts`
2. Generate React library or Workspace library `nx generate @nrwl/workspace:library test-lib --buildable`
3. Try to import the generated components into the app


### Nx Report

```shell
Node : 16.18.1
   OS   : darwin arm64
   yarn : 1.22.19
   
   nx : 15.3.3
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.3.3
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.3.3
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.3.3
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.3.3
   @nrwl/js : 15.3.3
   @nrwl/linter : 15.3.3
   @nrwl/nest : Not Found
   @nrwl/next : 15.3.3
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 15.0.2
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.3.3
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.3.3
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : 15.3.3
   @nrwl/webpack : 15.3.3
   @nrwl/workspace : 15.3.3
   typescript : 4.9.4
   ---------------------------------------
   Local workspace plugins:
   Unable to construct project graph
   ---------------------------------------
   Community plugins:

✨  Done in 0.56s.
```

**Comment by atanaskanchev** [comment]
No such issue with `15.2.4`

**Comment by ndcunningham** [comment]
Can you update your NX workspace and try the latest version?

**Comment by atanaskanchev** [comment]
Fixed by 15.4.0
